tenbilac/committee.py

```diff
@@ -0,0 +1,189 @@
+"""
+This is Tenbilac!
+This is a net (Wnet in the future) ensemble class.
+"""
+
+import numpy as np
+from multiprocessing import Process, Pipe, Queue
+from itertools import izip  
+
+import logging
+logger = logging.getLogger(__name__)
+
+from . import train
+from . import utils
+
+
+def spawn(f):  
+	def fun(pipe,x):  
+		pipe.send(f(x))  
+		pipe.close()  
+	return fun  
+
+def parmap(f, X, ncpu):  
+	"""
+	This is an alternative to multiprocessing.Pool to avoid the limitations of the package (pickling stuff...)
+	
```

Comment:
The function to map is not well defined. Having to call a instance method makes the whole thing much worse, especially since this called method does a lot of things and calls many other methods and functions. The function `map` pickles the data and gives it to a cpu along with the worker. Apparently, sometimes `map` does not pickle everything. I spent quite a long time finding a workaround.